lib/kube/proxy/auth.go

```diff
@@ -130,6 +130,9 @@ func getKubeCreds(ctx context.Context, log logrus.FieldLogger, tpClusterName, ku
 	res := make(map[string]*kubeCreds, len(cfg.Contexts))
 	// Convert kubeconfig contexts into kubeCreds.
 	for cluster, clientCfg := range cfg.Contexts {
+		if tlsConfig != nil {
+			clientCfg.ServerName = tlsConfig.ServerName
+		}
```

Comment:
Why are we overriding whatever is in client config? TLS routing relies on the server name.